cirq/sim/simulator.py

```diff
@@ -357,6 +356,19 @@ def _simulator_iterator(
         Yields:
             StepResults from simulating a Moment of the Circuit.
         """
+        param_resolver = param_resolver or study.ParamResolver({})
+        resolved_circuit = protocols.resolve_parameters(circuit, param_resolver)
+        _check_all_resolved(resolved_circuit)
+        actual_initial_state = 0 if initial_state is None else initial_state
+        return self._base_iterator(resolved_circuit, qubit_order, actual_initial_state)
+
+    @abc.abstractmethod
+    def _base_iterator(
```

Comment:
The docstring for this class should be updated to reflect the new requirement:
```
Implementors of this interface should implement the _base_iterator method.
```

Reply:
done